api-server/controllers/controllersv1/self_hosted.go

```diff
@@ -0,0 +1,157 @@
+	* 2. the token in request is the same as the token in config
+	* 3. There is no user in db
+	* 4. There is no org in db
+	* 5. There is no cluster in db
+	* If any of the above condition is not met, this route will return error
+	*
+	* this endpoint will:
+	* 1. create a user with admin permission,
+	* 2. create a org,and add the user to the org,
+	* 3. create a cluster, and add the org to the cluster,
+	 */
+
+	if config.YataiConfig.IsSass {
+		return nil, errors.New("admin user registration is not allowed in sass mode")
+	}
+	if schema.Token != config.YataiConfig.InitializationToken {
+		return nil, errors.New("invalid token")
+	}
+	_, total, err := services.UserService.List(ctx, services.ListUserOption{
+		BaseListOption: services.BaseListOption{
+			Start: utils.UintPtr(0),
+			Count: utils.UintPtr(1),
+		},
+		Order: utils.StringPtr("id ASC"),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "list users")
+	}
+	if total > 0 {
+		return nil, errors.New("default admin user already exists")
+	}
+	_, total, err = services.OrganizationService.List(ctx, services.ListOrganizationOption{
+		BaseListOption: services.BaseListOption{
+			Start: utils.UintPtr(0),
+			Count: utils.UintPtr(1),
+		},
+		Order: utils.StringPtr("id ASC"),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "list organizations")
+	}
+	if total > 0 {
+		return nil, errors.New("default organization already exists")
+	}
+
+	_, total, err = services.ClusterService.List(ctx, services.ListClusterOption{
+		BaseListOption: services.BaseListOption{
+			Start: utils.UintPtr(0),
+			Count: utils.UintPtr(1),
+		},
+		Order: utils.StringPtr("id ASC"),
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "list clusters")
+	}
+	if total > 0 {
+		return nil, errors.New("default cluster already exists")
+	}
+
+	// Create default admin user
+	adminUser, err := services.UserService.Create(ctx, services.CreateUserOption{
+		Name:      schema.Name,
+		FirstName: schema.FirstName,
+		LastName:  schema.LastName,
+		Email:     utils.StringPtrWithoutEmpty(schema.Email),
+		Password:  schema.Password,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "create user")
+	}
+
+	// Create default org
+	var defaultOrg *models.Organization
+	defaultOrg, err = services.OrganizationService.Create(ctx, services.CreateOrganizationOption{
+		CreatorId: adminUser.ID,
+		Name:      "default",
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "create default organization")
+	}
+	_, err = services.OrganizationMemberService.Create(ctx, adminUser.ID, services.CreateOrganizationMemberOption{
+		CreatorId:      adminUser.ID,
+		UserId:         adminUser.ID,
+		OrganizationId: defaultOrg.ID,
+		Role:           modelschemas.MemberRoleAdmin,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "create default organization member")
+	}
+
+	// create default cluster
+	var defaultCluster *models.Cluster
+	defaultCluster, err = services.ClusterService.Create(ctx, services.CreateClusterOption{
```

Comment:
update services/clustes.go, in reference of services/cluster_members.go. Use db transaction for create yatai component